### PR TITLE
Remove git update in build.sh

### PR DESCRIPTION
**Pull request by matinlotfali**
I think the git update commands in `build.sh` is causing inconsistencies. Typically, the user wants to prepare the repo beforehand and let the docker be built based on what is already visible. Right?

So I removed those lines and I made some dependency between git action jobs so it wouldn't start building a yocto image when the submodule check failed.

### CODE REVIEW

File: Build/build.sh [hunk 1]
```diff
@@ -1,6 +1,4 @@
 #!/bin/bash
-git submodule sync
-git submodule update --init --recursive
```
Comment (matinlotfali) [inline Build/build.sh:3]: @mistyron I mean this line

**Review by mistyron — APPROVED** [review]
LGTM

**Review by mistyron — APPROVED** [review]
LGTM

File: Build/build.sh (hunk 1 above)
Comment (mistyron) [reply Build/build.sh:3]: > @mistyron I mean this line

OK, yeah, good point!

**Pull request merged**